### ISSUE TITLE
Add support for Cropping layers

### ISSUE THREAD

**Issue by HamzaEzzRa** [fastmachinelearning/hls4ml#1308]
I have a [branch](https://github.com/HamzaEzzRa/hls4ml-custom-layers/tree/cropping) that adds support for Keras' [Cropping1D](https://keras.io/api/layers/reshaping_layers/cropping1d) and [Cropping2D](https://keras.io/api/layers/reshaping_layers/cropping2d) layers with Vivado and Vitis backend in both `io_parallel` and `io_stream` modes.

The implementation is straightforward and follows a similar style to zero-padding implementations. I added tests for Vivado and Vitis backends.

I have yet to run this on a board, but implementation was successful targeting a ZCU102, using Vivado 2019.1. Although this type of layer has a fairly niche usage, I still found a use case for it, and it could benefit the community.

**Comment by calad0i** [comment]
Close as completed with the merging #1309.
